### PR TITLE
2.8.x secfixes2

### PR DESCRIPTION
**Pull request by uncleflo**
See https://github.com/centreon/centreon/pull/6406

### CODE REVIEW

File: www/class/centreon-knowledge/procedures_DB_Connector.class.php [hunk 1]
```diff
@@ -111,10 +111,11 @@ public function toString() {
     public function query($query_string = NULL) {
 
     	if ($this->debug) {
-    		$query = str_replace("`", "", $query_string);
+    		/*$query = str_replace("`", "", $query_string);
```
Comment (sc979) [inline www/class/centreon-knowledge/procedures_DB_Connector.class.php]: please delete this comment if these lines are useless

File: www/include/Administration/corePerformance/processCommands.php [hunk 2]
```diff
@@ -97,7 +97,8 @@
     if (isset($command->localhostTab[$poller])) {
         shell_exec("sudo service" . $nagios_init_script . " start");
     } else {
-        shell_exec("echo 'START:".$poller."' >> $centcore_pipe");
+    	$command = escapeshellarg("START:".$poller."");
+        shell_exec("echo $command >> $centcore_pipe");
```
Comment (sc979) [inline www/include/Administration/corePerformance/processCommands.php]: indentation

File: www/class/centreonEscaping.class.php [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+	/**
+	 * Escape for Database Values. Alias for: CentreonDB::escape() (preferred, 
+	 *   as it is used everywhere already). The function has been created here for completeness. 
+	 *
+	 * For example, in case for:
+	 * SELECT * FROM hosts WHERE host_name='[INJECT UNTRUSTED STRING HERE]' OR host_alias LIKE '%[INJECT UNTRUSTED STRING HERE]%'
+	 *
+	 * Usage:
+	 * $sMyHostsQuery = "SELECT * FROM hosts WHERE host_name='".Esc::forQueryValue($sMyHostName)."' OR host_alias LIKE '%".Esc::forQueryValue($sMyHostName)."%'";
```
Comment (sc979) [inline www/class/centreonEscaping.class.php]: 120 chars limit -> please split this line and respect the PSR2

File: www/class/centreon-knowledge/procedures.class.php [hunk 4]
```diff
@@ -367,20 +367,20 @@ public function duplicate($template, $object, $type)
...
+                " VALUES ('0', '" . CentreonDB::escape($object) . "', '0', '0', '1', '" . $data["page_random"] . "', '" .
                 $dateTouch . "', '" . $data["page_latest"] . "', '" . $data["page_len"] . "')"
             );
             $DBRESULT = $this->DB->query("SELECT MAX(page_id) FROM " . $this->db_prefix . "page");
             $id = $DBRESULT->fetchRow();
 
             $this->DB->query(
                 "INSERT INTO `text` (old_id, old_text, old_flags) VALUE (NULL, '" .
-                $text["old_text"] . "', '" . $text["old_flags"] . "')"
+            		CentreonDB::escape($text["old_text"]) . "', '" . CentreonDB::escape($text["old_flags"]) . "')"
```
Comment (sc979) [inline www/class/centreon-knowledge/procedures.class.php]: indentation

File: www/class/centreon-knowledge/procedures.class.php [hunk 5]
```diff
@@ -367,20 +367,20 @@ public function duplicate($template, $object, $type)
...
             );
             $this->DB->query(
                 "INSERT INTO `revision` (rev_page, rev_text_id, rev_comment, rev_user_text, rev_timestamp, " .
                 "  rev_len) VALUE ('" . $id["MAX(page_id)"] . "', (SELECT MAX(old_id) FROM text), '" .
-                $revision["rev_comment"] . "', '" . $revision["rev_user_text"] . "','" . $dateTouch .
+            		CentreonDB::escape($revision["rev_comment"]) . "', '" . CentreonDB::escape($revision["rev_user_text"]) . "','" . $dateTouch .
```
Comment (sc979) [inline www/class/centreon-knowledge/procedures.class.php]: indentation

File: www/class/centreonEscaping.class.php [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+ * will correspond as much as possible the above quote, to make it as clear 
+ * and easy to understand as possible, when where and what to escape. For 
+ * example: To 'Escape for a Html Attribute', you use:
+ *  CentreonEscaping::forHtmlAttr(...) 
+ * or even better using the below shortcut class:
+ *  Esc::forHtmlAttr()
+ * 
+ * Good usage is to use these functions immediately inline with the surrounding 
+ * context. For example, in case of direct echoing $sHostName as a Host Name:
+ * <a href="<?php echo Esc::forHtmlAttr("main.php?p=20202&o=hd&host_name=".Esc::forUrlValue($sHostName)); ?>"><?php echo Esc::forHtmlContent($sHostName); ?></a>
```
Comment (sc979) [inline www/class/centreonEscaping.class.php]: please split in two lines < 120 chars

File: www/class/centreonEscaping.class.php [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+ * 
+ * A lot more information is available on OWASP site, and similar informations 
+ * are used in this class file.
+ * 
+ */
+class CentreonEscaping
+{
+
```
Comment (sc979) [inline www/class/centreonEscaping.class.php]: indentation -> 4 spaces

File: www/class/centreonEscaping.class.php [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+ */
+class CentreonEscaping
+{
+	
+	public function __construct()
+	{
+		
+	}
+	
+	//TODO: Need to have a look at the sanitizeShellString() function in /www/include/monitoring/external_cmd/functionsPopup.php and potentially add it here
```
Comment (sc979) [inline www/class/centreonEscaping.class.php]: 120 chars max per line + indentation for the whole paragraph

File: www/class/centreonEscaping.class.php [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+
+
+	/**
+     * Escape for Url Values
+     * 
+     * For example, in case for:
+     * http://www.forexample.com/folder/path/file.php?parameter=[INJECT UNTRUSTED STRING HERE]&anotherparameter=etc
+     * 
+     * Usage:
+     * $sMyUrl = "http://www.forexample.com/folder/path/file.php?parameter=".Esc::forUrlValue($sMyValue)."&anotherparameter=etc";
```
Comment (sc979) [inline www/class/centreonEscaping.class.php]: 120 chars max per line

File: www/class/centreonEscaping.class.php [hunk 10]
```diff
@@ -0,0 +1,159 @@
...
+	public function __construct()
+	{
+		
+	}
+	
+	//TODO: Need to have a look at the sanitizeShellString() function in /www/include/monitoring/external_cmd/functionsPopup.php and potentially add it here
+	
+
+
+	/**
```
Comment (sc979) [inline www/class/centreonEscaping.class.php]: indentation

File: www/class/centreonEscaping.class.php [hunk 11]
```diff
@@ -0,0 +1,159 @@
...
+     *
+     * @param $sValue string 
+     * @return string The escaped string, ready for injection in url parameter 
+     *   values (ie. AFTER the url ?, and BETWEEN the parameter = sign and 
+     *   BEFORE the next & parameter!)
+	 */
+	static public function forUrlValue($sValue) {
+		//URL-encode according to RFC 3986
+		return rawurlencode($sValue);
+		//There are two functions: rawurlencode() and urlencode(). urlencode() is used for posted data from a WWW form. Using the standards compliant function rawurlencode().
```
Comment (sc979) [inline www/class/centreonEscaping.class.php]: 120 chars max per line

File: www/class/centreonEscaping.class.php [hunk 12]
```diff
@@ -0,0 +1,159 @@
...
+	}
+
+	/**
+     * Escape for Html Attributes
+     * 
+     * For example, in case for:
+     * <div>[INJECT UNTRUSTED STRING HERE] <span>[INJECT UNTRUSTED STRING HERE]</span></div>
+     *
+     * Usage:
+     * $sMyHtmlContent = '<div>'.Esc::forHtmlContent($sMyContent).' <span>'.Esc::forHtmlContent($sMySubContent).'</span></div>';
```
Comment (sc979) [inline www/class/centreonEscaping.class.php]: 120 chars max per line + indentation for the whole paragraph

File: www/class/centreonEscaping.class.php [hunk 13]
```diff
@@ -0,0 +1,159 @@
...
+	}
+	
+	
+
+	/**
+	 * Escape for Database Values. Alias for: CentreonDB::escape() (preferred, 
+	 *   as it is used everywhere already). The function has been created here for completeness. 
+	 *
+	 * For example, in case for:
+	 * SELECT * FROM hosts WHERE host_name='[INJECT UNTRUSTED STRING HERE]' OR host_alias LIKE '%[INJECT UNTRUSTED STRING HERE]%'
```
Comment (sc979) [inline www/class/centreonEscaping.class.php]: 120 chars max per line + indentation for the whole paragraph

File: www/class/centreonEscaping.class.php (hunk 3 above)
Comment (sc979) [inline www/class/centreonEscaping.class.php]: 120 chars max per line + indentation for the whole paragraph

File: www/class/centreonLDAP.class.php [hunk 14]
```diff
@@ -774,6 +774,7 @@ private function getBindInfo($id)
     private function debug($msg)
     {
         if ($this->debugImport) {
+        	$msg = str_replace("\n", "\\n", str_replace("\r", "\\r", $msg));
```
Comment (sc979) [inline www/class/centreonLDAP.class.php]: indentation

File: www/class/centreonUtils.class.php [hunk 15]
```diff
@@ -56,9 +56,12 @@ public function visit($idPage, $redirect = true)
         } else {
             $http .= 'http://';
         }
-
+
```
Comment (sc979) [inline www/class/centreonUtils.class.php]: please remove white spaces

File: www/class/centreonUtils.class.php [hunk 16]
```diff
@@ -56,9 +56,12 @@ public function visit($idPage, $redirect = true)
         } else {
             $http .= 'http://';
         }
-
+        
         $newUrl = $http . $_SERVER['HTTP_HOST'] . $centreon->optGen["oreon_web_path"] . 'main.php?p=' . $idPage;
-
+        
+        if(!filter_var($newUrl, FILTER_VALIDATE_URL)) {
+        	return null;
```
Comment (sc979) [inline www/class/centreonUtils.class.php]: indentation

File: www/class/centreonUtils.class.php [hunk 17]
```diff
@@ -56,9 +56,12 @@ public function visit($idPage, $redirect = true)
         } else {
             $http .= 'http://';
         }
-
+        
         $newUrl = $http . $_SERVER['HTTP_HOST'] . $centreon->optGen["oreon_web_path"] . 'main.php?p=' . $idPage;
-
+        
+        if(!filter_var($newUrl, FILTER_VALIDATE_URL)) {
```
Comment (sc979) [inline www/class/centreonUtils.class.php]: PSR 2 -> a space is needed after the if

File: www/class/centreonWidget.class.php [hunk 18]
```diff
@@ -922,7 +923,7 @@ protected function insertWidgetPreferences($lastId, $config)
      */
     public function install($widgetPath, $directory)
     {
-        $config = $this->readConfigFile($widgetPath . "/" . $directory . "/configs.xml");
+        $config = $this->readConfigFile($widgetPath . "/".Esc::forUrlValue($directory)."/configs.xml");
```
Comment (sc979) [inline www/class/centreonWidget.class.php]: PSR2 -> a space is needed before and after the dots

File: www/include/Administration/corePerformance/processCommands.php [hunk 19]
```diff
@@ -97,7 +97,8 @@
     if (isset($command->localhostTab[$poller])) {
         shell_exec("sudo service" . $nagios_init_script . " start");
     } else {
-        shell_exec("echo 'START:".$poller."' >> $centcore_pipe");
+    	$command = escapeshellarg("START:".$poller."");
```
Comment (sc979) [inline www/include/Administration/corePerformance/processCommands.php]: a space is needed before and after the dots

File: www/include/configuration/configKnowledge/popup.php [hunk 20]
```diff
@@ -59,6 +59,8 @@
 require_once "centreonLog.class.php";
 require_once "centreonDB.class.php";
 
+require_once $centreon_path . "/www/class/centreonEscaping.class.php";
```
Comment (sc979) [inline www/include/configuration/configKnowledge/popup.php]: please remove the white space before this line

File: www/include/configuration/configKnowledge/popup.php [hunk 21]
```diff
@@ -82,11 +84,16 @@
 );
 
 if (isset($_GET["template"]) && $_GET["template"] != "") {
-    $proc->duplicate(
+    /*$proc->duplicate(
```
Comment (sc979) [inline www/include/configuration/configKnowledge/popup.php]: please delete this comment if these lines are useless

File: www/include/configuration/configKnowledge/proxy/proxy.php [hunk 22]
```diff
@@ -81,9 +82,9 @@
     header("Location: " . $proxy->url);
 } else {
     if (isset($_GET["host_name"]) && isset($_GET["service_description"])) {
-        header("Location: $WikiURL/?title=Service_:_" . $_GET["host_name"] . "_/_" . $_GET["service_description"]);
+        header("Location: $WikiURL/?title=Service_:_".Esc::forUrlValue($_GET["host_name"])."_/_" . Esc::forUrlValue($_GET["service_description"]));
```
Comment (sc979) [inline www/include/configuration/configKnowledge/proxy/proxy.php]: 120 chars max per line

File: www/include/configuration/configKnowledge/proxy/proxy.php [hunk 23]
```diff
@@ -81,9 +82,9 @@
...
     } else {
-        header("Location: $WikiURL/?title=Host_:_" . $_GET["host_name"]);
+        header("Location: $WikiURL/?title=Host_:_".Esc::forUrlValue($_GET["host_name"]) );
```
Comment (sc979) [inline www/include/configuration/configKnowledge/proxy/proxy.php]: please add a space before and after the dots

File: www/include/configuration/configObject/command/minPlayCommand.php [hunk 24]
```diff
@@ -66,7 +66,7 @@
 /* Replace HOSTADDRESS by the real content */
 while (preg_match("/@DOLLAR@HOSTADDRESS@DOLLAR@/", $resource_def, $matches) and $error_msg == "") {
     if (isset($_GET["command_hostaddress"]) && $_GET["command_hostaddress"] != "") {
-        $resource_def = str_replace("@DOLLAR@HOSTADDRESS@DOLLAR@", $_GET["command_hostaddress"], $resource_def);
+        $resource_def = str_replace("@DOLLAR@HOSTADDRESS@DOLLAR@", escapeshellarg($_GET["command_hostaddress"]), $resource_def);
```
Comment (sc979) [inline www/include/configuration/configObject/command/minPlayCommand.php]: 120 chars max per line

File: www/include/configuration/configObject/contact/ldapsearch.php [hunk 25]
```diff
@@ -49,7 +49,7 @@
     $oreon = $_SESSION["centreon"];
 }
 
-if (!isset($_POST['confList']) || !strlen($_POST['confList'])) {
+if (!isset($_POST['confList']) || !strlen($_POST['confList']) || !ctype_digit((string)trim(implode("", explode(",", $_POST['confList']))))) {
```
Comment (sc979) [inline www/include/configuration/configObject/contact/ldapsearch.php]: 120 chars max per line -> please split this line

File: www/include/core/header/header.php [hunk 26]
```diff
@@ -102,8 +102,10 @@ function microtime_float()
  * Get session and Check if session is not expired
  */
 $DBRESULT = $pearDB->query("SELECT `user_id` FROM `session` WHERE `session_id` = '".session_id()."'");
-if (!$DBRESULT->numRows()) {
-    header("Location: index.php?disconnect=2".$args);
+
+$newUrl = "index.php?disconnect=2".$args;
+if (!$DBRESULT->numRows() && filter_var($newUrl, FILTER_VALIDATE_URL)) {
+    header("Location: ".$newUrl);
```
Comment (sc979) [inline www/include/core/header/header.php]: please add space before and after the dots

File: www/include/core/header/header.php [hunk 27]
```diff
@@ -121,13 +123,21 @@ function microtime_float()
...
     } else {
         $args = null;
         foreach ($_GET as $key => $value) {
             $args ? $args .= "&".$key."=".$value : $args = $key."=".$value;
         }
-        header("Location: index.php?".$args."");
+        
+		$newUrl = "index.php?disconnect=1".$args;
+		if (filter_var($newUrl, FILTER_VALIDATE_URL)) {
+			header("Location: ".$newUrl);
```
Comment (sc979) [inline www/include/core/header/header.php]: PSR2 -> add a space before and after the dots

File: www/include/core/login/processLogin.php [hunk 28]
```diff
@@ -32,7 +32,9 @@
  * For more information : contact@centreon.com
  *
  */
-        
+
```
Comment (sc979) [inline www/include/core/login/processLogin.php]: please delete useless space

File: www/include/core/login/processLogin.php [hunk 29]
```diff
@@ -89,7 +91,7 @@
                 $minimize = '&min=1';
             }
             if (isset ($_GET["p"]) && $_GET["p"] != '') {
-                header('Location: main.php?p=' . $_GET["p"] . $minimize);
+                header('Location: main.php?p='.Esc::forUrlValue($_GET["p"]).$minimize);
```
Comment (sc979) [inline www/include/core/login/processLogin.php]: PSR2 -> add a space before and after the dots

File: www/include/home/customViews/widgetParam.php [hunk 30]
```diff
@@ -54,7 +54,7 @@
 $widgetObj = new CentreonWidget($centreon, $db);
 $title = "";
 $defaultTab = array();
-if (isset($_REQUEST['view_id']) && $_REQUEST['view_id'] && isset($_REQUEST['widget_id']) && $_REQUEST['widget_id']) {
+if (isset($_REQUEST['view_id']) && $_REQUEST['view_id'] && ctype_digit($_REQUEST['view_id']) && isset($_REQUEST['widget_id']) && $_REQUEST['widget_id'] && ctype_digit($_REQUEST['widget_id'])) {
```
Comment (sc979) [inline www/include/home/customViews/widgetParam.php]: please split this line -> 120 chars max per line

File: www/include/options/accessLists/groupsACL/DB-Func.php [hunk 31]
```diff
@@ -68,7 +68,7 @@ function testGroupExistence($name = null)
     }
     $query = "SELECT acl_group_id, acl_group_name "
         . "FROM acl_groups "
-        . "WHERE acl_group_name = '" . htmlentities($name, ENT_QUOTES, "UTF-8")."' ";
+        . "WHERE acl_group_name = '" . htmlentities($name, ENT_QUOTES, "UTF-8")."' "; //Seriously ???? CentreonDB::escape($name)
```
Comment (sc979) [inline www/include/options/accessLists/groupsACL/DB-Func.php]: 120 chars max per line  AND please remove useless comments
the code isn't a chat box -> contact us using our mails, our slack community or using github comments feature... but not like this

File: www/include/reporting/dashboard/ajaxReporting_js.php [hunk 32]
```diff
@@ -34,18 +34,22 @@
  */
     
 require_once realpath(dirname(__FILE__) . "/../../../../config/centreon.config.php");
+require_once './class/centreonEscaping.class.php';
 
 if ($type == "Service") {
-    $arg = "id=".$service_id."&host_id=".$host_id;
+    $arg = "id=".Esc::forUrlValue($service_id)."&host_id=".Esc::forUrlValue($host_id);
```
Comment (sc979) [inline www/include/reporting/dashboard/ajaxReporting_js.php]: PSR2 -> please add a space before and after the dots

File: www/include/reporting/dashboard/ajaxReporting_js.php [hunk 33]
```diff
@@ -34,18 +34,22 @@
...
 } else {
-    $arg = "id=".$id;
+    $arg = "id=".Esc::forUrlValue($id);
```
Comment (sc979) [inline www/include/reporting/dashboard/ajaxReporting_js.php]: please add a space before and after the dots

File: www/include/reporting/dashboard/ajaxReporting_js.php [hunk 34]
```diff
@@ -34,18 +34,22 @@
...
 }
 
-$arg .= "&color[UP]=".$oreon->optGen["color_up"]."&color[UNDETERMINED]=".$oreon->optGen["color_undetermined"].
-        "&color[DOWN]=".$oreon->optGen["color_down"]."&color[UNREACHABLE]=".$oreon->optGen["color_unreachable"].
-        "&color[OK]=".$oreon->optGen["color_ok"]."&color[WARNING]=".$oreon->optGen["color_warning"].
-        "&color[CRITICAL]=".$oreon->optGen["color_critical"]."&color[UNKNOWN]=".$oreon->optGen["color_unknown"];
-$arg = str_replace("#", "%23", $arg);
+$arg .= "&color[UP]=".Esc::forUrlValue($oreon->optGen["color_up"]).
```
Comment (sc979) [inline www/include/reporting/dashboard/ajaxReporting_js.php]: PSR2 -> please add a space before and after the dots
AND please split this line (120 chars max per line)

File: www/include/reporting/dashboard/viewHostGroupLog.php [hunk 35]
```diff
@@ -52,10 +52,9 @@
 /*
  *  Getting hostgroup to report
  */
-isset($_GET["item"]) ? $id = $_GET["item"] : $id = "NULL";
-isset($_POST["item"]) ? $id = $_POST["item"] : $id;
+isset($_GET["item"]) && (ctype_digit(trim($_GET["item"])) || strtoupper(trim($_GET["item"])) == "NULL") ? $id = trim($_GET["item"]) : $id = "NULL";
```
Comment (sc979) [inline www/include/reporting/dashboard/viewHostGroupLog.php]: 120 chars max

File: www/include/reporting/dashboard/viewHostGroupLog.php [hunk 36]
```diff
@@ -52,10 +52,9 @@
...
+isset($_POST["item"]) && (ctype_digit(trim($_POST["item"])) || strtoupper(trim($_POST["item"])) == "NULL") ? $id = trim($_POST["item"]) : $id;
```
Comment (sc979) [inline www/include/reporting/dashboard/viewHostGroupLog.php]: 120 chars max

File: www/include/reporting/dashboard/viewHostLog.php [hunk 37]
```diff
@@ -49,8 +49,8 @@
 /*
  *  Getting host to report
  */
-isset($_GET["host"]) ? $id = $_GET["host"] : $id = "NULL";
-isset($_POST["host"]) ? $id = $_POST["host"] : htmlentities($id, ENT_QUOTES, "UTF-8");
+isset($_GET["host"]) && (ctype_digit(trim($_GET["host"])) || strtoupper(trim($_GET["host"])) == "NULL") ? $id = trim($_GET["host"]) : $id = "NULL";
```
Comment (sc979) [inline www/include/reporting/dashboard/viewHostLog.php]: 120 chars max

File: www/include/reporting/dashboard/viewHostLog.php [hunk 38]
```diff
@@ -49,8 +49,8 @@
...
+isset($_POST["host"]) && (ctype_digit(trim($_POST["host"])) || strtoupper(trim($_POST["host"])) == "NULL") ? $id = trim($_POST["host"]) : $id;
```
Comment (sc979) [inline www/include/reporting/dashboard/viewHostLog.php]: 120 chars max

File: www/include/reporting/dashboard/viewServicesGroupLog.php [hunk 39]
```diff
@@ -45,10 +45,9 @@
 /*
  *  Getting servic egroup to report
  */
-isset($_GET["item"]) ? $id = $_GET["item"] : $id = "NULL";
-isset($_POST["item"]) ? $id = $_POST["item"] : $id;
+isset($_GET["item"]) && (ctype_digit(trim($_GET["item"])) || strtoupper(trim($_GET["item"])) == "NULL") ? $id = trim($_GET["item"]) : $id = "NULL";
```
Comment (sc979) [inline www/include/reporting/dashboard/viewServicesGroupLog.php]: 120 chars max

File: www/include/reporting/dashboard/viewServicesGroupLog.php [hunk 40]
```diff
@@ -45,10 +45,9 @@
...
+isset($_POST["item"]) && (ctype_digit(trim($_POST["item"])) || strtoupper(trim($_POST["item"])) == "NULL") ? $id = trim($_POST["item"]) : $id;
```
Comment (sc979) [inline www/include/reporting/dashboard/viewServicesGroupLog.php]: 120 chars max

File: www/include/reporting/dashboard/viewServicesLog.php [hunk 41]
```diff
@@ -45,10 +45,10 @@
 /*
  *  Getting service to report
  */
-isset($_GET["host_id"]) ? $host_id = $_GET["host_id"] : $host_id = "NULL";
-isset($_POST["host_id"]) ? $host_id = $_POST["host_id"] : $host_id;
-isset($_GET["item"]) ? $service_id = $_GET["item"] : $service_id = "NULL";
-isset($_POST["item"]) ? $service_id = $_POST["item"] : $service_id;
+isset($_GET["host_id"]) && (ctype_digit(trim($_GET["host_id"])) || strtoupper(trim($_GET["host_id"])) == "NULL") ? $host_id = trim($_GET["host_id"]) : $host_id = "NULL";
```
Comment (sc979) [inline www/include/reporting/dashboard/viewServicesLog.php]: 120 chars max

File: www/include/reporting/dashboard/viewServicesLog.php [hunk 42]
```diff
@@ -45,10 +45,10 @@
...
+isset($_POST["host_id"]) && (ctype_digit(trim($_POST["host_id"])) || strtoupper(trim($_POST["host_id"])) == "NULL") ? $host_id = trim($_POST["host_id"]) : $host_id;
```
Comment (sc979) [inline www/include/reporting/dashboard/viewServicesLog.php]: 120 chars max

File: www/include/reporting/dashboard/viewServicesLog.php [hunk 43]
```diff
@@ -45,10 +45,10 @@
...
+isset($_GET["item"]) && (ctype_digit(trim($_GET["item"])) || strtoupper(trim($_GET["item"])) == "NULL") ? $service_id = trim($_GET["item"]) : $service_id = "NULL";
```
Comment (sc979) [inline www/include/reporting/dashboard/viewServicesLog.php]: 120 chars max

File: www/include/reporting/dashboard/viewServicesLog.php [hunk 44]
```diff
@@ -45,10 +45,10 @@
...
+isset($_POST["item"]) && (ctype_digit(trim($_POST["item"])) || strtoupper(trim($_POST["item"])) == "NULL") ? $service_id = trim($_POST["item"]) : $service_id;
```
Comment (sc979) [inline www/include/reporting/dashboard/viewServicesLog.php]: 120 chars max

File: www/include/reporting/dashboard/viewServicesLog.php [hunk 45]
```diff
@@ -147,8 +147,10 @@
      */
     $tpl->assign(
         "link_csv_url",
-        "./include/reporting/dashboard/csvExport/csv_ServiceLogs.php?host="
-        .$host_id."&service=".$service_id."&start=".$start_date."&end=".$end_date
+        "./include/reporting/dashboard/csvExport/csv_ServiceLogs.php?host=".Esc::forUrlValue($host_id).
+    		"&service=".Esc::forUrlValue($service_id).
```
Comment (sc979) [inline www/include/reporting/dashboard/viewServicesLog.php]: indentation AND 120 chars max

File: www/include/views/graphs/getGraphAjax.php [hunk 46]
```diff
@@ -112,6 +112,7 @@ function getGraphByService($host, $svcId, $title, $isAdmin, $lca)
 /* By hostgroups */
 if (isset($_POST['host_group_filter'])) {
     foreach ($_POST['host_group_filter'] as $hgId) {
+    	if(!ctype_digit((string)$hgId)) continue;
```
Comment (sc979) [inline www/include/views/graphs/getGraphAjax.php]: PSR2 -> please add a space after the if

File: www/include/views/graphs/getGraphAjax.php [hunk 47]
```diff
@@ -121,6 +122,7 @@ function getGraphByService($host, $svcId, $title, $isAdmin, $lca)
 /* By hosts */
 if (isset($_POST['host_selector'])) {
     foreach ($_POST['host_selector'] as $host) {
+    	if(!ctype_digit((string)$host)) continue;
```
Comment (sc979) [inline www/include/views/graphs/getGraphAjax.php]: please add a space after the if

File: www/include/views/graphs/getGraphAjax.php [hunk 48]
```diff
@@ -131,6 +133,7 @@ function getGraphByService($host, $svcId, $title, $isAdmin, $lca)
         $services = getMyServiceGroupServices($sgId);
         foreach ($services as $hostSvcId => $svcName) {
             list($hostId, $svcId) = explode('_', $hostSvcId);
+            if(!ctype_digit((string)$hostId) || !ctype_digit((string)$svcId)) continue;
```
Comment (sc979) [inline www/include/views/graphs/getGraphAjax.php]: please add a space after the if

File: www/include/views/graphs/getGraphAjax.php [hunk 49]
```diff
@@ -140,6 +143,7 @@ function getGraphByService($host, $svcId, $title, $isAdmin, $lca)
 if (isset($_POST['service_selector'])) {
     foreach ($_POST['service_selector'] as $selectedService) {
         list($hostId, $svcId) = explode('-', $selectedService['id']);
+        if(!ctype_digit((string)$hostId) || !ctype_digit((string)$svcId)) continue;
```
Comment (sc979) [inline www/include/views/graphs/getGraphAjax.php]: please add a space after the if

File: www/include/views/graphs/graphStatus/displayServiceStatus.php [hunk 50]
```diff
@@ -289,7 +289,8 @@ function getStatusDBDir($pearDBO)
     $command_line = escape_command("$command_line");
 
     if ($centreon->optGen["debug_rrdtool"] == "1") {
-        error_log("[" . date("d/m/Y H:s") ."] RDDTOOL : $command_line \n", 3, $centreon->optGen["debug_path"]."rrdtool.log");
+    	$msg = str_replace("\n", "\\n", str_replace("\r", "\\r", $command_line));
+        error_log("[" . date("d/m/Y H:s") ."] RDDTOOL : $msg \n", 3, $centreon->optGen["debug_path"]."rrdtool.log");
```
Comment (sc979) [inline www/include/views/graphs/graphStatus/displayServiceStatus.php]: indentation AND please add a space before and after the dots AND 120 chars per line max

**Comment by uncleflo** [comment]
Updates pushed, please check?

**Comment by uncleflo** [comment]
@sc979 I apologize that I missed some? Thank you for helping.

File: www/include/configuration/configObject/command/minPlayCommand.php (hunk 24 above)
Comment (sc979) [reply www/include/configuration/configObject/command/minPlayCommand.php]: done

File: www/include/configuration/configKnowledge/proxy/proxy.php (hunk 23 above)
Comment (sc979) [reply www/include/configuration/configKnowledge/proxy/proxy.php]: done

File: www/include/configuration/configKnowledge/proxy/proxy.php (hunk 22 above)
Comment (sc979) [reply www/include/configuration/configKnowledge/proxy/proxy.php]: done

File: www/include/configuration/configKnowledge/popup.php (hunk 21 above)
Comment (sc979) [reply www/include/configuration/configKnowledge/popup.php]: done

File: www/include/configuration/configKnowledge/popup.php (hunk 20 above)
Comment (sc979) [reply www/include/configuration/configKnowledge/popup.php]: done

File: www/class/centreonWidget.class.php (hunk 18 above)
Comment (sc979) [reply www/class/centreonWidget.class.php]: done

File: www/class/centreonUtils.class.php (hunk 17 above)
Comment (sc979) [reply www/class/centreonUtils.class.php]: done

File: www/class/centreonUtils.class.php (hunk 15 above)
Comment (sc979) [reply www/class/centreonUtils.class.php]: done

File: www/include/configuration/configObject/contact/ldapsearch.php (hunk 25 above)
Comment (sc979) [reply www/include/configuration/configObject/contact/ldapsearch.php]: done

File: www/include/core/header/header.php (hunk 26 above)
Comment (sc979) [reply www/include/core/header/header.php]: done

File: www/include/core/header/header.php (hunk 27 above)
Comment (sc979) [reply www/include/core/header/header.php]: done

File: www/include/core/login/processLogin.php (hunk 28 above)
Comment (sc979) [reply www/include/core/login/processLogin.php]: done

File: www/include/core/login/processLogin.php (hunk 29 above)
Comment (sc979) [reply www/include/core/login/processLogin.php]: done

File: www/include/options/accessLists/groupsACL/DB-Func.php (hunk 31 above)
Comment (sc979) [reply www/include/options/accessLists/groupsACL/DB-Func.php]: done

File: www/include/home/customViews/widgetParam.php (hunk 30 above)
Comment (sc979) [reply www/include/home/customViews/widgetParam.php]: done

File: www/include/reporting/dashboard/ajaxReporting_js.php (hunk 32 above)
Comment (sc979) [reply www/include/reporting/dashboard/ajaxReporting_js.php]: done

File: www/include/reporting/dashboard/ajaxReporting_js.php (hunk 33 above)
Comment (sc979) [reply www/include/reporting/dashboard/ajaxReporting_js.php]: done

File: www/include/reporting/dashboard/ajaxReporting_js.php (hunk 34 above)
Comment (sc979) [reply www/include/reporting/dashboard/ajaxReporting_js.php]: done

File: www/include/reporting/dashboard/viewHostGroupLog.php (hunk 35 above)
Comment (sc979) [reply www/include/reporting/dashboard/viewHostGroupLog.php]: done

File: www/include/reporting/dashboard/viewHostGroupLog.php (hunk 36 above)
Comment (sc979) [reply www/include/reporting/dashboard/viewHostGroupLog.php]: done

File: www/include/reporting/dashboard/viewHostLog.php (hunk 37 above)
Comment (sc979) [reply www/include/reporting/dashboard/viewHostLog.php]: done

File: www/include/reporting/dashboard/viewHostLog.php (hunk 38 above)
Comment (sc979) [reply www/include/reporting/dashboard/viewHostLog.php]: done

**Comment by sc979** [comment]
hi uncleflo.
I've made most of the PSR2 corrections. As I can't push on your repo, you need to cherry-pick my commits from here -> 
https://github.com/centreon/centreon/pull/6434
**Commits sha:** 
1de5545eae4183f1a1bb5a383f51afe74fd414ca / 4bb5e83e84426b33ba24ede62699077565f3522e
5d67cf3c22abf53190c6ea2fdccfadc41ec53a6b / d5a33efbf77aaae73d5b7c66faebfbe527276dea
1d6b58bfa548bd5ebb9d048d9ad02d3ab270866e
**or** to make a new PR including these commits, using the following command lines :
" git fetch origin
git checkout -b uncleflo-2.8.x-secfixes2 origin/uncleflo-2.8.x-secfixes2 "

**Comment by cgagnaire** [comment]
Moved to #6434

**Pull request closed without merging**